admin/code/LeftAndMain.php

```diff
@@ -355,10 +356,12 @@ protected function getSchemaResponse($form) {
 	 * based on the request data.
 	 *
 	 * @param Form $form
+	 * @param String $id Optional, will default to the current request URL
 	 * @return array
 	 */
-	protected function getSchemaForForm(Form $form) {
+	protected function getSchemaForForm(Form $form, $id = null) {
 		$request = $this->getRequest();
+		$id = $id ? $id : $request->getURL();
```

Comment:
Best case is that the same schema is downloaded for every record of a given type, and that dynamic behaviour is done within JavaScript. Not every form will be able to be built this way, but it will be a use-case that will be applied sometimes.

This doesn't preclude us from making id mandatory - it seems like it's usually set to a URL regardless.